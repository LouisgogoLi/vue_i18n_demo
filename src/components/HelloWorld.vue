<template>
  <div class="hello">
    <h1>{{ t("Welcome_to_Your_Vue", { vueMessage: "vue.js" }) }}</h1>
    <p>
      {{ $t("configure_customize_project") }}<br />
      {{ $t("check_out_the") }}
      <a href="https://cli.vuejs.org" target="_blank" rel="noopener"
        >vue-cli documentation</a
      >.
    </p>
    <h3>{{ $t("installed_cli_plugins") }}</h3>
    <h4>{{ props.propsMsg }}</h4>
    <ul>
      <li>
        <a
          href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel"
          target="_blank"
          rel="noopener"
          >babel</a
        >
      </li>
      <li>
        <a
          href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-router"
          target="_blank"
          rel="noopener"
          >router</a
        >
      </li>
      <li>
        <a
          href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-vuex"
          target="_blank"
          rel="noopener"
          >vuex</a
        >
      </li>
      <li>
        <a
          href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint"
          target="_blank"
          rel="noopener"
          >eslint</a
        >
      </li>
    </ul>
    <h3>{{ $t("essential_links") }}</h3>
    <ul>
      <li>
        <a href="https://vuejs.org" target="_blank" rel="noopener">Core Docs</a>
      </li>
      <li>
        <a href="https://forum.vuejs.org" target="_blank" rel="noopener"
          >Forum</a
        >
      </li>
      <li>
        <a href="https://chat.vuejs.org" target="_blank" rel="noopener"
          >Community Chat</a
        >
      </li>
      <li>
        <a href="https://twitter.com/vuejs" target="_blank" rel="noopener"
          >Twitter</a
        >
      </li>
      <li>
        <a href="https://news.vuejs.org" target="_blank" rel="noopener">News</a>
      </li>
    </ul>
    <h3>{{ $t("ecosystem") }}</h3>
    <ul>
      <li>
        <a href="https://router.vuejs.org" target="_blank" rel="noopener"
          >vue-router</a
        >
      </li>
      <li>
        <a href="https://vuex.vuejs.org" target="_blank" rel="noopener">vuex</a>
      </li>
      <li>
        <a
          href="https://github.com/vuejs/vue-devtools#vue-devtools"
          target="_blank"
          rel="noopener"
          >vue-devtools</a
        >
      </li>
      <li>
        <a href="https://vue-loader.vuejs.org" target="_blank" rel="noopener"
          >vue-loader</a
        >
      </li>
      <li>
        <a
          href="https://github.com/vuejs/awesome-vue"
          target="_blank"
          rel="noopener"
          >awesome-vue</a
        >
      </li>
    </ul>
  </div>
</template>

<script>
import { useI18n } from "vue-i18n";
export default {
  name: "HelloWorld",
  props: {
    propsMsg: {
      type: String,
      default: "沒有傳值",
    },
  },
  setup(props) {
    const { t } = useI18n({
      inheritLocale: true,
      useScope: "local",
    });

    // Something todo ..

    return { t, props };
  },
};
</script>

<i18n>
{
  "zh_tw": {
    "Welcome_to_Your_Vue": "歡迎到{vueMessage}專案"
  },
  "en": {
    "Welcome_to_Your_Vue": "Welcome to Your {vueMessage} App"
  },
  "ja": {
    "Welcome_to_Your_Vue": "{vueMessage}プロジェクトへようこそ"
  },
}
</i18n>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
