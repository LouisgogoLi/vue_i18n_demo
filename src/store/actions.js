export default {
  handSetLanguageState({ commit }, payload) {
    commit("SetLanguageState", payload);
  },
};
