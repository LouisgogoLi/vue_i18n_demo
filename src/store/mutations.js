export default {
  SetLanguageState(state, payload) {
    state.language = payload;
  },
};
