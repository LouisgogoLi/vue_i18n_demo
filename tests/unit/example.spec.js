import { shallowMount } from "@vue/test-utils";
import HomeView from "@/views/HomeView.vue";
import HelloWorld from "@/components/HelloWorld.vue";

import { createI18n } from "vue-i18n";

const i18n = createI18n({
  legacy: false,
});

import store from "@/store";

describe("HelloWorld.vue", () => {
  it("renders props.msg when passed", () => {
    const propsMsg = "new message";
    const wrapper = shallowMount(HelloWorld, {
      props: { propsMsg },
      global: {
        plugins: [i18n],
      },
    });
    expect(wrapper.text()).toMatch(propsMsg);
  });
});

describe("HomeView.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(HomeView, {
      global: {
        plugins: [i18n, store],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
