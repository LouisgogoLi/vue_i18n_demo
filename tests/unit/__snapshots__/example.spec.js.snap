// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView.vue renders correctly 1`] = `
<div class="home">
  <div><select>
      <option value="zh_tw">中文</option>
      <option value="en">English</option>
      <option value="ja">日本語</option>
    </select></div><img alt="Vue logo" src="">
  <hello-world-stub propsmsg="測試vue"></hello-world-stub>
</div>
`;
